redwood-treehouse-composeui/src/main/kotlin/app/cash/redwood/treehouse/composeui/TreehouseContent.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood.treehouse.composeui
+
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.foundation.layout.Box
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberUpdatedState
+import app.cash.redwood.treehouse.HostConfiguration
+import app.cash.redwood.treehouse.TreehouseApp
+import app.cash.redwood.treehouse.TreehouseView
+import app.cash.redwood.widget.compose.ComposeWidgetChildren
+import kotlinx.coroutines.flow.MutableStateFlow
+
+@Composable
+fun <T : Any> TreehouseContent(
+  treehouseApp: TreehouseApp<T>,
+  content: TreehouseView.Content<T>,
+) {
+  val hostConfiguration = HostConfiguration(
+    darkMode = isSystemInDarkTheme(),
+  )
+
+  val rememberedContent = rememberUpdatedState(content)
+  val treehouseView = remember {
+    object : TreehouseView<T> {
+      override val boundContent: TreehouseView.Content<T> get() = rememberedContent.value
```

Comment:
I'm having a hard time tracing all this code without context. Just to confirm, this is never cached by a caller?